src/features/DisplayControls/Diagnosis/components/Diagnosis.jsx

```diff
@@ -0,0 +1,139 @@
+import React, { useEffect } from "react";
+import {
+  DataTable,
+  TableCell,
+  TableHead,
+  TableHeader,
+  TableBody,
+  Table,
+  TableRow,
+  DataTableSkeleton,
+} from "carbon-components-react";
+import { useState } from "react";
+import { FormattedMessage } from "react-intl";
+import { getPatientDiagnosis } from "../utils/DiagnosisUtils";
+import PropTypes from "prop-types";
+import "../styles/Diagnosis.scss";
+
+const Diagnosis = (props) => {
+   
+    const {patientId} = props;
+    const[diagnosis, setDiagnosis] = useState([]);
+    const[isLoading, setIsLoading] = useState(true);
+
+    const NoDiagnosisMessage = (
+    <FormattedMessage
+        id={"NO_DIAGNOSIS_MESSAGE"}
+        defaultMessage={"No Diagnosis found for this patient"}
+    />
+    );
+  
+    const diagnosisHeaders = [
+    {
+        id: '1',
+        header: 'Diagnosis',
+        key: 'diagnosis',
+        isSortable: false 
+    },
+    {
+        id: '2',
+        header: 'Order',
+        key: 'order',
+        isSortable: true 
+    },
+    {
+        id: '3',
+        header: 'Certainty',
+        key: 'certainty',
+        isSortable: true 
+    },
+    {
+        id: '4',
+        header: 'Status',
+        key: 'status',
+        isSortable: true 
+    },
+    {
+        id: '5',
+        header: 'Diagnosed By',
+        key: 'diagnosedBy',
+        isSortable: false 
+    },
+    {
+        id: '6',
+        header: 'Diagnosis Date ',
+        key: 'diagnosisDate',
+        isSortable: true 
+    },
+    ];
+
+    const mapDiagnosisData = (diagnosisList) => {
+        const mappedDiagnoses = diagnosisList === undefined ? [] : diagnosisList.map((diagnosis) => {
+            let status = diagnosis.diagnosisStatusConcept ? "Inactive" : "Active";
+            let diagnosisDate = new Date(diagnosis.diagnosisDateTime).toLocaleDateString();
```

Comment:
what is the date format? I think we are following DD MMM YYYY, please check and if that's the case create a common utility method in utils for formatting date and time and use it in here.

Reply:
added a date util file